### PR TITLE
31 spearbit

### PR DESCRIPTION
**Pull request by LayneHaber**
## Description

Fixes https://github.com/spearbit-audits/connext/issues/31

## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## High-level change(s) description - from the user's perspective

- Removes native asset handling for transfers

## Related Issue(s)

Fixes https://github.com/spearbit-audits/connext/issues/31

### CODE REVIEW

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 1]
```diff
@@ -80,8 +74,9 @@ contract BridgeFacet is BaseConnextFacet {
   event XCalled(
     bytes32 indexed transferId,
     uint256 indexed nonce,
+    uint256 bridgedAmt,
     XCallArgs xcallArgs,
-    XCalledEventArgs args,
+    address bridged,
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:79]: can we reorder to have bridged and bridgedAmt next to each other? also can we rename to bridgedAmount and bridgedAsset?

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:79]: or maybe bridgedAssetId to match args.transactingAssetId

File: packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol [hunk 2]
```diff
@@ -53,36 +53,28 @@ library AssetLogic {
 
   /**
    * @notice Handles transferring funds from msg.sender to the Connext contract.
-   * @dev If using the native asset, will automatically wrap
+   * @dev Only allows native asset to be usef for fee
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:56]: usef

**Review by jakekidd — APPROVED** [review]
looks good

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 3]
```diff
@@ -76,26 +70,35 @@ contract BridgeFacet is BaseConnextFacet {
...
+   * necessary.
+   * @param caller - The account that called the function.
    */
   event XCalled(
     bytes32 indexed transferId,
     uint256 indexed nonce,
     XCallArgs xcallArgs,
-    XCalledEventArgs args,
+    address bridgedAsset,
+    uint256 bridgedAmount,
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:87]: @LayneHaber fyi I changed the order and names here

**Review by jakekidd — CHANGES_REQUESTED** [review]
Looks like `Executor` is still checking for native asset on `execute`

**Review by liu-zhipeng — APPROVED** [review]

File: packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol [hunk 4]
```diff
@@ -53,36 +53,28 @@ library AssetLogic {
...
-      // When transferring native asset to the contract, always make sure that the
-      // asset is properly wrapped
-      wrapNativeAsset(_assetAmount);
-      _assetId = address(s.wrapper);
-    } else {
-      if (msg.value != _fee) revert AssetLogic__handleIncomingAsset_ethWithErcTransfer();
-
-      // Transfer asset to contract
-      transferAssetToContract(_assetId, _assetAmount);
+      revert AssetLogic__handleIncomingAsset_nativeAssetNotSupported();
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:71]: isn't this redundant with the xcall sanity check?

Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:69]: no because this isn't only used in xcall. but for xcall it is, yes

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:69]: ah hm sad

**Review by jakekidd — APPROVED** [review]

File: packages/deployments/contracts/contracts/core/connext/facets/AssetFacet.sol [hunk 5]
```diff
@@ -166,6 +138,9 @@ contract AssetFacet is BaseConnextFacet {
     address _adoptedAssetId,
     address _stableSwapPool
   ) external onlyOwner {
+    // Native asset support does not exist in this contract
```
Comment (0xbok) [inline packages/deployments/contracts/contracts/core/connext/facets/AssetFacet.sol:141]: Can't comment on the Natspec description since it's not part of the PR, but you can remove line 126-128 since native asset is not allowed anymore.

File: packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol [hunk 6]
```diff
@@ -96,42 +86,17 @@ library AssetLogic {
     address _assetId,
```
Comment (0xbok) [inline packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:86]: update Natspec for native asset.

File: packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol [hunk 7]
```diff
@@ -247,15 +217,13 @@ contract Executor is IExecutor {
...
    * @param _assetId - Asset associated with call
    * @param _to - Where call was attempted
    * @param _recovery - Where to send funds
    * @param _amount - Amount to send
    */
   function _sendToRecovery(
-    bool _isNative,
     bool _hasIncreased,
     address _assetId,
     address payable _to,
```
Comment (0xbok) [inline packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol:229]: i think `payable` keyword can be removed now.

File: packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol [hunk 8]
```diff
@@ -247,15 +217,13 @@ contract Executor is IExecutor {
    * @notice Sends funds to the specified recovery address
    * @dev Called if the external call data fails, it's not a contract, or the amount in native
```
Comment (0xbok) [inline packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol:218]: native asset is not handled now, comment can be updated.

File: packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 [
   "error AssetFacet__addAssetId_alreadyAdded()",
-  "error AssetFacet__addAssetId_noWrapperExists()",
+  "error AssetFacet__addAssetId_nativeAsset()",
   "error AssetFacet__removeAssetId_notAdded()",
   "error AssetFacet__setTokenRegistry_invalidTokenRegistry()",
   "error AssetFacet__setWrapper_invalidWrapper()",
```
Comment (gpersoon) [inline packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json:6]: AssetFacet__setWrapper_invalidWrapper non longer used

File: packages/deployments/contracts/abi/contracts/core/connext/interfaces/IConnextHandler.sol/IConnextHandler.json [hunk 10]
```diff
@@ -127,5 +129,5 @@
   "function unpause()",
   "function withdrawSwapAdminFees(bytes32)",
   "function wrapper() view returns (address)",
```
Comment (gpersoon) [inline packages/deployments/contracts/abi/contracts/core/connext/interfaces/IConnextHandler.sol/IConnextHandler.json:131]: wrapper() no longer used

File: packages/deployments/contracts/abi/contracts/core/connext/interfaces/IConnextHandler.sol/IConnextHandler.json [hunk 11]
```diff
@@ -127,5 +129,5 @@
...
-  "function xcall(tuple(tuple(address,bytes,uint32,uint32,address,address,bool,bool,address,uint256,uint256,uint256),address,uint256)) payable returns (bytes32)"
+  "function xcall(tuple(tuple(address,bytes,uint32,uint32,address,address,bool,bool,address,uint256,uint256,uint256),address,uint256,uint256)) payable returns (bytes32)"
```
Comment (gpersoon) [inline packages/deployments/contracts/abi/contracts/core/connext/interfaces/IConnextHandler.sol/IConnextHandler.json:132]: wrapper() no longer used (line 131)

File: packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json [hunk 12]
```diff
@@ -1,6 +1,6 @@
 [
   "error AssetFacet__addAssetId_alreadyAdded()",
-  "error AssetFacet__addAssetId_noWrapperExists()",
+  "error AssetFacet__addAssetId_nativeAsset()",
```
Comment (gpersoon) [inline packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json:3]: AssetFacet__setWrapper_invalidWrapper no longer used (line 6)

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 13]
```diff
@@ -295,18 +298,18 @@ contract BridgeFacet is BaseConnextFacet {
...
-   * case, custody of the local assets will be kept here.
-   *
    * @param _args - The XCallArgs arguments.
    * @return bytes32 - The transfer ID of the newly created crosschain transfer.
    */
   function xcall(XCallArgs calldata _args) external payable nonReentrant whenNotPaused returns (bytes32) {
     // Sanity checks.
     bytes32 remoteInstance;
     {
+      // Not native asset
```
Comment (gpersoon) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:308]: Function `xcall()` doesn't have to be `payable` anymore?

File: packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol [hunk 14]
```diff
@@ -53,36 +53,26 @@ library AssetLogic {
 
   /**
    * @notice Handles transferring funds from msg.sender to the Connext contract.
-   * @dev If using the native asset, will automatically wrap
+   * @dev Only allows native asset to be used for fee
```
Comment (gpersoon) [inline packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:56]: comment about `@param _fee` not entirely accurate, all of native asset is meant for for

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:56]: @LayneHaber can we remove the `_fee` param here entirely? as well as the check below?

File: packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol (hunk 4 above)
Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol:62]: resolved! `xcall` now calls `transferAssetToContract` directly instead of this method

File: packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json (hunk 12 above)
Comment (LayneHaber) [reply packages/deployments/contracts/abi/contracts/core/connext/facets/AssetFacet.sol/AssetFacet.json:3]: These files will all be regenerated (and are autogenerated)
